crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -50,16 +51,27 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
         let db = &*state.store;
         let merchant_id = &merchant_account.merchant_id;
         let storage_scheme = merchant_account.storage_scheme;
-        let (mut payment_intent, mut payment_attempt, currency, amount, connector_response);
+        let (currency, amount, connector_response);
 
         let payment_id = payment_id
             .get_payment_intent_id()
             .change_context(errors::ApiErrorResponse::PaymentNotFound)?;
 
-        payment_intent = db
+        // Stage 1
+
+        let payment_intent_fut = db
             .find_payment_intent_by_payment_id_merchant_id(&payment_id, merchant_id, storage_scheme)
-            .await
-            .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)?;
+            .map(|x| x.change_context(errors::ApiErrorResponse::PaymentNotFound));
+
+        let mandate_details_fut = helpers::get_token_pm_type_mandate_details(
+            state,
+            request,
+            mandate_type.clone(),
+            merchant_account,
+        );
+
+        let (mut payment_intent, mandate_details) =
+            futures::try_join!(payment_intent_fut, mandate_details_fut)?;
```

Comment:
`helpers::get_token_pm_type_mandate_details`. This function internally does two database calls, and when joined with payment_intent db call, which has only one db call, to get the result the max time would be two db call.

Can you try doing a tokio::spawn for `get_token_pm_type_mandate_details`?

Reply:
Instead internally, we could make the calls concurrent. This we create a 3 node tree for resolution.

Reply:
also, doing tokio spawn will make it so that, if any failure occurs in it, it will not affect the flow, which shouldn't be the case

Reply:
seems like its not possible to internally parallelize this, as the calls are hierarchically

Reply:
You can catch the error by awaiting on the join handle. These are just read queries, so if there is any error we can catch it there.

Reply:
If we await the join handle, then that's exactly equivalent to what's happening here

Reply:
What I'm telling is to await only for that join handle and not club it with payment_intent fetch